sql/core/src/test/resources/sql-tests/inputs/ansi/conditional-functions.sql

```diff
@@ -1,6 +1,41 @@
 -- Tests for conditional functions
-CREATE TABLE t USING PARQUET AS SELECT c1, c2 FROM VALUES(1, 0),(2, 1) AS t(c1, c2);
+--SET spark.sql.maxMetadataStringLength = 500
+
+CREATE TABLE t USING PARQUET AS SELECT c1, c2 FROM VALUES(1d, 0),(2d, 1),(null, 1),(CAST('NaN' AS DOUBLE), 0) AS t(c1, c2);
 
 SELECT nanvl(c1, c1/c2 + c1/c2) FROM t;
+-- do not fail at compile side
```

Comment:
```suggestion
-- do not fail during query compilation
```

Reply:
For testing this kind of detail, I think a unit test is better. Can we add a few test cases in `ConstantFoldingSuite`?

Reply:
Then we don't need to test EXPLAIN here and set `spark.sql.maxMetadataStringLength`

Reply:
moved some tests to `ConstantFoldingSuite`